ttnn/cpp/ttnn/operations/matmul/device/matmul_op.cpp

```diff
@@ -1113,6 +1114,13 @@ void Matmul::validate(
         a_shape[-1],
         b_shape[-2]);
 
+    const bool is_optional_output_tensor_empty = !optional_output_tensors.empty() && !optional_output_tensors.at(0).has_value();
```

Comment:
This variable name and logic is too complicated and doesn't look correct.
```
empty(). has_value() Result
F             F                    T. -> if (!is...) = F - not executed
F             T                    F -> if (!is...) = T - executed
T             N/A                F -> if(!is...) = T - executed - at(0) should throw
```
What happened when you executed this for all 3 combinations?